CHANGELOG.md

```diff
@@ -2,6 +2,85 @@
+- Changed `commonOptions` in `AdapterRead` from a getter to a function. [[#374](https://github.com/juttle/juttle/issues/374)]
+- Removed the read/write shims for legacy adapters. [[#370](https://github.com/juttle/juttle/pull/370)]
+- Made `Program#deactivate` not delete `this.graph`. [[#400](https://github.com/juttle/juttle/pull/400)]
+- Cleaned up filter AST: [[#319](https://github.com/juttle/juttle/issues/319)]
+  - Replaced `ReducerCall` by `FunctionCall`.
+  - Renamed `NumericLiteral` to `NumberLiteral`.
+  - Renamed `RegularExpressionLiteral` to `RegExpLiteral`.
+  - Renamed `PropertyAccess` to `MemberExpression`.
+  - Renamed `FunctionCall` to `CallExpression`.
+  - Renamed the `value` property of `RegExpLiteral` to `pattern`.
+  - Renamed the `text` property of `FilterLiteral` to `source`.
+  - Renamed the `base` property of `MemberExpression` to `object`.
+  - Renamed the `name` property of `MemberExpression` to `property`.
+  - Renamed the `name` property of `CallExpression` to `callee`.
+  - Renamed the `expression` property of `UnaryExpression` to `argument`.
+  - Renamed the `condition` property of `ConditionalExpression` to `test`.
+  - Renamed the `trueExpression` property of `ConditionalExpression` to `alternate`.
+  - Renamed the `falseExpression` property of `ConditionalExpression` to `consequent`.
+- Simplified filter AST: [[#418](https://github.com/juttle/juttle/pull/418)]
+  - Removed `ExpressionFilterTerm` nodes (which the compiler typically just passes-through).
+  - Replaced `SimpleFilterTerm` nodes which contain `StringLiteral` (and thus represent FTS) with structurally simpler `FulltextFilterTerm` nodes.
+  - Replaced `SimpleFilterTerm` nodes which contain `FilterLiteral` (and thus represent embedded filter expression coming from an input) with the actual `FilterLiteral` AST (effectively inlining the literal).
+- Simplified AST representation of fields. [[#318](https://github.com/juttle/juttle/issues/318)]
+- Introduced `ASTTransformer` and `StaticFilterCompiler` classes. The latter is a base class the filter compilers in adapters shuld be generally derived from. [[#420](https://github.com/juttle/juttle/pull/420), [#446](https://github.com/juttle/juttle/pull/446)]
+- Converted the codebase to use ES6 classes instead of `extendable-base`. [[#341](https://github.com/juttle/juttle/issues/341)]
+- Replaced Backbone.js Events and built-in Node.js events with EventEmitter3. [[#49](https://github.com/juttle/juttle/issues/49)]
+- Moved classes representing Juttle types (currently `JuttleMoment` and `Filter`) to `lib/runtime/types`. [[#312](https://github.com/juttle/juttle/pull/312)]
+- Finished consolidation of Juttle runtime: [[#97](https://github.com/juttle/juttle/issues/97)]
+  - Dissolved the global `Juttle` variable.
+  - Made compiled programs independent on `JuttleMoment` and `Filter` variables.
```

Comment:
Grammatically I think this should either be "independent of" or "not depend on". Same for the next one.